python/ray/experimental/data/datasource/file_based_datasource.py

```diff
@@ -208,6 +210,13 @@ def _unwrap_protocol(path):
     return parsed.netloc + parsed.path
 
 
+def _maybe_wrap_fs(filesystem: "pyarrow.fs.FileSystem"):
```

Comment:
```suggestion
def _wrap_s3_serialization_workaround(filesystem: "pyarrow.fs.FileSystem"):
```

(and reference the GH issue)